dialogue-serde-core/src/main/java/com/palantir/conjure/java/dialogue/serde/core/ConjureErrorDecoder.java

```diff
@@ -51,17 +52,18 @@
  * {@link RemoteException}) if a {@link RemoteException} could not be extracted, e.g., when the given {@link
  * Response} does not adhere to an expected format.
  */
-enum ErrorDecoder {
-    INSTANCE;
+public final class ConjureErrorDecoder implements ErrorDecoder {
```

Comment:
Instead of implementing an interface that's only used by the annotation processor, could the processor provide this interface, and an adapter from the default implementation into the annotaiton-processor error handler? That way we're free to add data from the Runtime object to the default dialogue ErrorDecoder as long as we can provide sane default args from the annotation processor shim.

Reply:
Yeah, just need to make the enum public and that should work.

Reply:
Done